testing/state-transition/state-transition.go

```diff
@@ -129,21 +138,23 @@ func SetupTestState(t *testing.T, cs chain.Spec) (
 		depositStore,
 		mocksSigner,
 		func(bytes.B48) ([]byte, error) {
-			return dummyProposerAddr, nil
+			return DummyProposerAddr, nil
 		},
 		nodemetrics.NewNoOpTelemetrySink(),
 	)
 
+	// by default we keep checks at minimum. It is up
+	// to single tests to redefine the ctx along their needs.
 	ctx := transition.NewTransitionCtx(
-		context.Background(),
+		ms,
 		0, // time
-		dummyProposerAddr,
+		DummyProposerAddr,
 	).
 		WithVerifyPayload(false).
 		WithVerifyRandao(false).
 		WithVerifyResult(false).
 		WithMeterGas(false).
 		WithOptimisticEngine(true)
 
-	return sp, beaconState, depositStore, ctx
+	return sp, beaconState, depositStore, ctx, cms, execEngine
```

Comment:
we need to export the mocked index because we need to set expectations right when handling the payload